fetch_raw_data.py

```diff
@@ -157,17 +157,27 @@ def fetch_from_recovery_csv(user, google_cloud_credentials_file_path, raw_data_d
         log.info(f"Exported TracedData")
 
 def fetch_listening_groups_csvs(google_cloud_credentials_file_path, pipeline_configuration, raw_data_dir):
+
+    uploaded_listening_group_files = [file.split("/")[-1] for file in google_cloud_utils.list_blobs(google_cloud_credentials_file_path,
+                                                                   pipeline_configuration.project_dataset_bucket_url,
+                                                                   pipeline_configuration.listening_group_bucket_path)]
+
     for k, v in pipeline_configuration.listening_group_csv_urls.items():
         for i, listening_group_csv_url in enumerate(v):
-            listening_group = listening_group_csv_url.split("/")[-1]
+            listening_group_csv = listening_group_csv_url.split("/")[-1]
+
+            if listening_group_csv not in uploaded_listening_group_files:
+                log.info(f"{listening_group_csv}' not yet uploaded to {pipeline_configuration.project_dataset_bucket_url};"
```

Comment:
I think this is overkill, if the file can't be found, just print the path that failed, no need to worry about explicitly knowing the bucket name. Having a project_dataset_bucket_url means we (i) have even more configuration we need to set, when the aim should be to have less, (ii) have introduced the requirement that all listening groups need be kept in the same bucket (which is probably ok, but not a constraint we need to introduce).

Reply:
updated to use try-except pattern. Thanks.